src/map/map.ts

```diff
@@ -1138,84 +1008,53 @@ export class PropertiesMap {
-        } else {
-            this._options.x.min.value = layout.xaxis.range[0];
-            this._options.x.max.value = layout.xaxis.range[1];
+        this._options.x.min.value = layout.xaxis.range[0];
+        this._options.x.max.value = layout.xaxis.range[1];
 
-            this._options.y.min.value = layout.yaxis.range[0];
-            this._options.y.max.value = layout.yaxis.range[1];
+        this._options.y.min.value = layout.yaxis.range[0];
+        this._options.y.max.value = layout.yaxis.range[1];
 
-            this._updateAllMarkers();
-        }
-    }
-
-    /**
-     * Update the position of the given marker.
-     *
-     * In 3D mode, the markers uses the second Plotly trace.
-     * In 2D mode, these markers are HTML div styled as colored circles that
-     * we manually move around, saving a call to `restyle`.
-     *
-     * @param data data of the marker to update
-     */
-    private _updateSelectedMarker(data: MarkerData): void {
-          const selected = data.current;
-          const marker = data.marker;
-
-          if (this._is3D()) {
-              // we have to update all symbols at the same time
-              this._updateAll3DMarkers();
-          } else {
-              const xaxis = this._plot._fullLayout.xaxis;
-              const yaxis = this._plot._fullLayout.yaxis;
-
-              const computeX = (value: number) => xaxis.l2p(value) + xaxis._offset;
-              const computeY = (value: number) => yaxis.l2p(value) + yaxis._offset;
-
-              const rawX = this._coordinates(this._options.x, 0) as number[][];
-              const rawY = this._coordinates(this._options.y, 0) as number[][];
-
-              const x = computeX(rawX[0][selected]);
-              const y = computeY(rawY[0][selected]);
-
-              // hide the point if it is outside the plot, allow for up to 10px
-              // overflow (for points just on the border)
-              const minX = computeX(xaxis.range[0]) - 10;
-              const maxX = computeX(xaxis.range[1]) + 10;
-              const minY = computeY(yaxis.range[1]) - 10;
-              const maxY = computeY(yaxis.range[0]) + 10;
-              if (!isFinite(x) || !isFinite(y) || x < minX || x > maxX || y < minY || y > maxY) {
-                  marker.style.display = 'none';
-              } else {
-                  marker.style.display = 'block';
-              }
-
-              marker.style.top = `${y}px`;
-              const plotWidth = this._plot.getBoundingClientRect().width;
-              marker.style.right = `${plotWidth - x}px`;
-          }
+        if (!this._is3D()) {this._updateMarkers(); }
     }
 
     /**
      * Update the position, color & size of all markers in the map
      */
-    private _updateAllMarkers(): void {
+    private _updateMarkers(data: MarkerData[] = Array.from(this._selected.values())): void {
         if (this._is3D()) {
+            for (const datum of data) {
```

Comment:
could also be `data.forEach((d) => d.toggleVisible(false))` if you want

Reply:
Yes, yes I do.